src/app/common/authentication/use-legacy-auth-bitcoin-addresses.ts

```diff
@@ -0,0 +1,35 @@
+import { bytesToHex } from '@noble/hashes/utils';
+
+import { useNativeSegwitNetworkSigners } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+import { useTaprootNetworkSigners } from '@app/store/accounts/blockchain/bitcoin/taproot-account.hooks';
+
+export function useGetLegacyAuthBitcoinAddresses() {
+  const deriveAllNativeSegWitNetworkSigners = useNativeSegwitNetworkSigners();
+  const deriveAllTaprootNetworkSigners = useTaprootNetworkSigners();
+
+  return (accountIndex: number) => {
```

Comment:
`useCallback`?

Reply:
Often feel like it's an over optimisation.

Reply:
I used to always use `useCallback` and `useMemo` until I read this: 
https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
IMO these hooks exemplify everything that's wrong with react. Signals remove the need for these entirely.

Reply:
Ok, thx for feedback, I was often unsure of this too but thought it was asked for in past CR of my own work. I'll keep this in mind moving fwd. 👍

Reply:
I probably suggested using it in another PR 😊